core/shared/src/main/scala/bobcats/Hash1.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bobcats
+
+import fs2.Stream
+import scodec.bits.ByteVector
+
+/**
+ * Hash for a single algorithm.
+ *
+ * Use this class if you have a specific `HashAlgorithm` known in advance or you're using a
+ * customized algorithm not covered by the `HashAlgorithm` class.
+ */
+sealed trait Hash1[F[_]] {
```

Comment:
The idea is that this should wrap the `MessageDigest`/`EVP_MD`/`Hash` equivalent on each platform so folks can construct them when the `HashAlgorithm` isn't complete.

It's also means that you can fail-fast if you have an application which is dependent on certain algorithm's, which imo is better than waiting until later.